tests/app/Infrastructure/Controller/GetsWalletBalanceControllerTest.php

```diff
@@ -5,48 +5,22 @@
 use App\Domain\DataSources\CoinDataSource;
 use App\Domain\DataSources\WalletDataSource;
 use App\Domain\Wallet;
+use App\Infrastructure\ApiServices\CoinloreApiService;
+use App\Infrastructure\Persistence\ApiCoinDataSource;
 use Illuminate\Support\Facades\Cache;
 use Mockery;
 use Tests\TestCase;
 
 class GetsWalletBalanceControllerTest extends TestCase
 {
-    private WalletDataSource $walletDataSource;
-
-    protected function setUp(): void
-    {
-        parent::setUp();
-        $this->walletDataSource = Mockery::mock(WalletDataSource::class);
-        $this->app->bind(WalletDataSource::class, function () {
-            return $this->walletDataSource;
-        });
-    }
-
     /**
      * @test
      */
-    public function ifBadWalletIdThrowsBadRequest()
-    {
-        $this->walletDataSource
-            ->expects("findById")
-            ->with(null)
-            ->times(0)
-            ->andReturn(null);
-
-        $response = $this->get('api/wallet/-5/balance');
-        $response->assertBadRequest();
-    }
-
-    /**
-     * @test
-     */
-    public function ifWalletIdNotFoundThrowsError()
+    public function throwsErrorWhenWalletIdNotFound()
```

Comment:
Think what is gonna happen in the test:
walletIdWasNotFound

And think why:
walletDoesNotExist

walletIdWasNotFoundIfwalletDoesNotExist?